http.c

```diff
@@ -353,10 +360,49 @@ evhttp_method(enum evhttp_cmd_type type)
 		method = "MOVE";
 		break;
 	default:
-		method = NULL;
+		/* setup the structure to allow for the cmp.
+		 *
+		 * if the cmp function is set, it has the ability to
+		 * modify method and flags. Other fields will be
+		 * ignored.
+		 *
+		 * NOTE: the flags returned are OR'd with the current
+		 *       flags.
+		 */
+		ext_method.method = NULL;
+		ext_method.type   = type;
+		ext_method.flags  = tmp_flags;
+
+		if (evcon->ext_method_cmp != NULL &&
+			evcon->ext_method_cmp(&ext_method) == 0) {
+
+		    if (ext_method.type != type) {
```

Comment:
In this case, at least `method` should be set to `NULL` for now

Reply:
I'm returning NULL now